Modules/posixmodule.c

```diff
@@ -5703,13 +5746,17 @@ os_spawnv_impl(PyObject *module, int mode, path_t *path, PyObject *argv)
     }
     argvlist[argc] = NULL;
 
+#if !defined(HAVE_RTPSPAWN)
     if (mode == _OLD_P_OVERLAY)
         mode = _P_OVERLAY;
+#endif
 
     Py_BEGIN_ALLOW_THREADS
     _Py_BEGIN_SUPPRESS_IPH
 #ifdef HAVE_WSPAWNV
     spawnval = _wspawnv(mode, path->wide, argvlist);
+#elif defined(HAVE_RTPSPAWN)
+    spawnval = _rtp_spawn(mode, path->narrow, (const char **)argvlist, NULL);
```

Comment:
is the cast really needed? Isn't EXECV_CHAR type the char type on VxWorks?

Reply:
Needed. As I commented above for the prototype of rtpSpawn(), both argv and envp are const qualified pointers. The casting here is used to suppress the build warnings.

Reply:
Ok